src/components/picker/types.tsx

```diff
@@ -0,0 +1,229 @@
+  /**
+   * Custom picker props (when using renderPicker, will apply on the button wrapper)
+   */
+  customPickerProps?: ExpandableOverlayProps;
+  /**
+   * Add onPress callback for when pressing the picker
+   */
+  onPress?: () => void;
+  /**
+   * @deprecated
+   * A function that extract the unique value out of the value prop in case value has a custom structure (e.g. {myValue, myLabel})
+   */
+  getItemValue?: (value: PickerValue) => any;
+  /**
+   * @deprecated
+   * A function that extract the label out of the value prop in case value has a custom structure (e.g. {myValue, myLabel})
+   */
+  getItemLabel?: (value: PickerValue) => string;
+  /**
+   * A function that returns the label to show for the selected Picker value
+   */
+  getLabel?: (value: PickerValue) => string;
+  /**
+   * The picker modal top bar props
+   */
+  topBarProps?: ModalTopBarProps;
+  /**
+   * Show search input to filter picker items by label
+   */
+  showSearch?: boolean;
+  /**
+   * Style object for the search input (only when passing showSearch)
+   */
+  searchStyle?: PickerSearchStyle;
+  /**
+   * Placeholder text for the search input (only when passing showSearch)
+   */
+  searchPlaceholder?: string;
+  /**
+   * callback for picker modal search input text change (only when passing showSearch)
+   */
+  onSearchChange?: (searchValue: string) => void;
+  /**
+   * Render custom search input (only when passing showSearch)
+   */
+  renderCustomSearch?: (props: PickerItemsListProps) => React.ReactElement;
+  /**
+   * Allow to use the native picker solution (different style for iOS and Android)
+   */
+  useNativePicker?: boolean;
+  /**
+   * Callback for rendering a custom native picker inside the dialog (relevant to native picker only)
+   */
+  renderNativePicker?: () => React.ReactElement;
+  /**
+   * Pass props to the list component that wraps the picker options (allows to control FlatList behavior)
+   */
+  listProps?: FlatListProps<any>;
+  /**
+   * Pass props to the picker modal
+   */
+  pickerModalProps?: object;
+  /**
+   * Custom container style
+   */
+  containerStyle?: StyleProp<ViewStyle>;
+  /**
+   * Callback for modal onShow event
+   */
+  onShow?: () => void;
+  /**
+   * Component test id
+   */
+  testID?: string;
+}
+
+export interface PickerPropsWithSingle extends PickerBaseProps {
+  mode: PickerModes.SINGLE;
+  value: PickerSingleValue;
+}
+
+export interface PickerPropsWithMulti extends PickerBaseProps {
+  mode: PickerModes.MULTI;
+  value: PickerMultiValue;
+}
+
+export type PickerProps = PickerPropsWithSingle | PickerPropsWithMulti;
+
+export interface PickerItemProps {
+  /**
+   * Item's value
+   */
+  value: PickerSingleValue;
+  /**
+   * Item's label
+   */
+  label: string;
+  /**
+   * Item's label style
+   */
+  labelStyle?: StyleProp<TextStyle>;
+  /**
+   * Custom function for the item label (e.g (value) => customLabel)
+   */
+  getItemLabel: PickerProps['getItemLabel'];
```

Comment:
Is `getItemLabel` prop not deprecated here as well?

Reply:
Not at the moment. 
Anyway, the whole migration of `getItemLabel` is not relevant to this PR